src/Shared/Model/PackageExistence/PackageVersionExistence.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Model.PackageExistence;
+
+using Contracts;
+using Enums;
+using System.Collections.Generic;
+
+/// <summary>
+/// Represents a package version that currently exists.
+/// </summary>
+public record PackageVersionExists : IPackageExistence
+{
+    public bool Exists => true;
+    public bool HasEverExisted => true;
+}
+
+/// <summary>
+/// Represents a package version that never existed.
+/// </summary>
+public record PackageVersionNotFound : IPackageExistence
+{
+    public bool Exists => false;
+    public bool HasEverExisted => false;
+}
+
+/// <summary>
+/// Represents a package version that was removed, and why.
+/// </summary>
+/// <param name="RemovalReasons">The reasons (if any) for why a package was removed.</param>
+public record PackageVersionRemoved(IReadOnlySet<PackageVersionRemovalReason> RemovalReasons) : PackageVersionNotFound
+{
+    public new bool HasEverExisted => true;
```

Comment:
I don't think you want to use new here.

The value of `HasEverExisted` will vary depending on the type of the variable, so if these are cast as `PackageVersionNotFound` objects they will actually end up with `HasEverExisted` false from the base `PackageVersionNotFound`. 

For Example:

![image](https://user-images.githubusercontent.com/98900/190046377-79803dc7-bd16-48e8-8ad8-4e95b2d83255.png)

Reply:
Thanks for explaining! I didn't really have a good understanding of what new did, it's just what my IDE suggested I do.

Reply:
Done!

Reply:
👍. I think new is pretty confusing, I don't think I've yet found the case where it's actually what I want.